platform/viewer/src/customHooks/useServer.js

```diff
@@ -32,28 +35,68 @@ const getServers = (appConfig, project, location, dataset, dicomStore) => {
       wadoRoot: pathUrl,
     };
     servers = GoogleCloudUtilServers.getServers(data, dicomStore);
+    if (!isValidServer(servers[0], appConfig)) {
+      return;
+    }
   }
 
   return servers;
 };
 
-const updateServer = (
+const isValidServer = (server, appConfig) => {
+  if (appConfig.enableGoogleCloudAdapter) {
+    return GoogleCloudUtilServers.isValidServer(server);
+  }
+
+  return !!server;
+};
+
+const setServers = (dispatch, servers) => {
+  const action = {
+    type: 'SET_SERVERS',
+    servers,
+  };
+  dispatch(action);
+};
+
+const useServerFromUrl = (
+  servers = [],
+  previousServers,
+  activeServer,
+  urlBasedServers,
   appConfig,
-  dispatch,
   project,
   location,
   dataset,
   dicomStore
 ) => {
-  const servers = getServers(appConfig, project, location, dataset, dicomStore);
+  // update state from url available only when gcloud on
+  if (!appConfig.enableGoogleCloudAdapter) {
+    return false;
+  }
 
-  if (servers && servers.length) {
-    const action = {
-      type: 'SET_SERVERS',
-      servers,
-    };
-    dispatch(action);
+  const serverHasChanged = previousServers !== servers && previousServers;
+
+  // do not update from url. use state instead.
+  if (serverHasChanged) {
+    return false;
+  }
+
+  // if no valid urlbased servers
+  if (!urlBasedServers || !urlBasedServers.length) {
+    return false;
+  } else if (!servers.length || !activeServer) {
+    // no current valid server
+    return true;
   }
+
+  const newServer = urlBasedServers[0];
+
+  let exists = _.some(servers, server => {
```

Comment:
es6 Arrays can do this already, so you can remove Lodash for this: https://youmightnotneed.com/lodash/#some